src/components/NavBar/SearchBar.tsx

```diff
@@ -0,0 +1,546 @@
+        }
+      }
+    }
+
+    document.addEventListener('keydown', keyDownHandler)
+
+    return () => {
+      document.removeEventListener('keydown', keyDownHandler)
+    }
+  }, [toggleOpen, hoveredIndex, totalSuggestions])
+
+  return (
+    <Box className={styles.searchBarDropdown}>
+      {hasInput ? (
+        // Empty or Up to 8 combined tokens and nfts
+        <Column gap="20">
+          {isNFTPage ? (
+            <>
+              {collectionSearchResults}
+              {tokenSearchResults}
+            </>
+          ) : (
+            <>
+              {tokenSearchResults}
+              {collectionSearchResults}
+            </>
+          )}
+        </Column>
+      ) : (
+        // Recent Searches, Trending Tokens, Trending Collections
+        <Column gap="20">
+          {searchHistory.length > 0 && (
+            <SearchBarDropdownSection
+              hoveredIndex={hoveredIndex}
+              startingIndex={0}
+              setHoveredIndex={setHoveredIndex}
+              toggleOpen={toggleOpen}
+              suggestions={searchHistory}
+              header={'Recent searches'}
+              headerIcon={<ClockIcon />}
+            />
+          )}
+          {(trendingTokens?.length ?? 0) > 0 && !isNFTPage && (
+            <SearchBarDropdownSection
+              hoveredIndex={hoveredIndex}
+              startingIndex={searchHistory.length}
+              setHoveredIndex={setHoveredIndex}
+              toggleOpen={toggleOpen}
+              suggestions={trendingTokens ?? []}
+              header={'Popular tokens'}
+              headerIcon={<TrendingArrow />}
+            />
+          )}
+          {(trendingCollections?.length ?? 0) > 0 && !isTokenPage && (
+            <SearchBarDropdownSection
+              hoveredIndex={hoveredIndex}
+              startingIndex={searchHistory.length + (isNFTPage ? 0 : trendingTokens?.length ?? 0)}
+              setHoveredIndex={setHoveredIndex}
+              toggleOpen={toggleOpen}
+              suggestions={trendingCollections as unknown as GenieCollection[]}
+              header={'Trending NFT collections'}
+              headerIcon={<TrendingArrow />}
+            />
+          )}
+        </Column>
+      )}
+    </Box>
+  )
+}
+
+function isCollection(suggestion: GenieCollection | FungibleToken | TrendingCollection) {
+  return (suggestion as FungibleToken).decimals === undefined
+}
+
+export const SearchBar = () => {
+  const [isOpen, toggleOpen] = useReducer((state: boolean) => !state, false)
+  const [searchValue, setSearchValue] = useState('')
+  const debouncedSearchValue = useDebounce(searchValue, 300)
+  const searchRef = useRef<HTMLDivElement>(null)
+  const location = useLocation()
+  const { width: windowWidth } = useWindowSize()
+
+  useOnClickOutside(searchRef, () => {
+    isOpen && toggleOpen()
+  })
+
+  // TODO NFT Search Results implmented here
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
rm?

Reply:
This is needed for initializing a blank collection array. This won't be needed when we've added collection search results